pkg/exec/builder/action.go

```diff
@@ -73,20 +75,22 @@ func unmarshalActionMap(actionMap map[string][]interface{}, builder BuildableAct
 //		 err := yaml.Unmarshal(contents, &action)
 //		 return &action, err
 //	 })
-func LoadAction(cxt *portercontext.Context, commandFile string, unmarshal func([]byte) (interface{}, error)) error {
-	contents, err := readInputFromStdinOrFile(cxt, commandFile)
+func LoadAction(ctx context.Context, porterCtx *portercontext.Context, commandFile string, unmarshal func([]byte) (interface{}, error)) error {
+	//lint:ignore SA4006 ignore unused ctx for now
+	ctx, span := tracing.StartSpan(ctx)
+	defer span.EndSpan()
+
+	contents, err := readInputFromStdinOrFile(porterCtx, commandFile)
 	if err != nil {
-		return err
+		return span.Error(err)
 	}
 
 	result, err := unmarshal(contents)
-	if cxt.Debug {
-		fmt.Fprintf(cxt.Err, "DEBUG Parsed Input:\n%#v\n", result)
-	}
 	if err != nil {
-		return fmt.Errorf("could not unmarshal input:\n %s: %w", string(contents), err)
+		return span.Error(fmt.Errorf("could not unmarshal input:\n %s: %w", string(contents), err))
 	}
 
+	span.Debugf("Parsed Mixin Input:\n%#v", result)
```

Comment:
Will this have potential to print out sensitive data?

Reply:
I had removed some other prints because of the same problem and must have missed this. If you see more elsewhere, let's make sure to remove debug statements that could be printing out injected creds/params!